File: Factory/Factory/Objective-C/MonsterFactory.m
//
//  MonsterFactory.m
//  Factory
//
//  Created by LMinh on 8/9/18.
//  Copyright © 2018 LMinh. All rights reserved.
//

#import "MonsterFactory.h"

@implementation MonsterFactory

+ (Monster *)createMonsterWithType:(MonsterType)type {
    switch (type) {
        case MonsterType_Lion:
            return [Lion new];
        case MonsterType_Dragon:
            return [Dragon new];
        case MonsterType_Wolf:
            return [Wolf new];
    }
}

@end

File: Factory/Factory/Swift/MonsterFactory.swift
//
//  MonsterFactory.swift
//  Factory
//
//  Created by LMinh on 8/9/18.
//  Copyright © 2018 LMinh. All rights reserved.
//

import Foundation

enum MonsterType: Int {
    case Lion = 1
    case Dragon
    case Wolf
}

class MonsterFactory {
     static func createMonster(type: MonsterType) -> Monster {
        switch type {
        case .Lion:
            return Lion()
        case .Dragon:
            return Dragon()
        case .Wolf:
            return Wolf()
        }
    }
}
